From 6a11c77b8a494ecd7ce46ea1f670252735a4d19e Mon Sep 17 00:00:00 2001
From: Minghuan Lian <Minghuan.Lian@freescale.com>
Date: Fri, 3 Jul 2015 14:57:08 +0800
Subject: [PATCH 037/124] irqchip/gicv3-its: fix getting its pointer

Structure its_node is referenced by the host_data of the MSI
domain's parent.

Signed-off-by: Minghuan Lian <Minghuan.Lian@freescale.com>
---
 drivers/irqchip/irq-gic-v3-its.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-gic-v3-its.c b/drivers/irqchip/irq-gic-v3-its.c
index 2ef95dc..54f0092 100644
--- a/drivers/irqchip/irq-gic-v3-its.c
+++ b/drivers/irqchip/irq-gic-v3-its.c
@@ -1290,7 +1290,8 @@ static int its_msi_prepare(struct irq_domain *domain, struct device *dev,
 
 	dev_dbg(dev, "ITT %d entries, %d bits\n", nvec, ilog2(nvec));
 	dev_id = PCI_DEVID(pdev->bus->number, pdev->devfn);
-	return __its_msi_prepare(domain->parent, dev_alias.dev_id, dev, dev_alias.count, info);
+	return __its_msi_prepare(domain, dev_alias.dev_id,
+				 dev, dev_alias.count, info);
 }
 
 static struct msi_domain_ops its_pci_msi_ops = {
-- 
2.1.0.27.g96db324

